File: 1786-count-the-number-of-consistent-strings/1786-count-the-number-of-consistent-strings.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) 
    {
        int consistent = 0;
        unordered_map<char, int> hashset;
        int flag = 1;
        for(int i = 0; i < allowed.length(); i++)
        {
            hashset[allowed[i]]++;
        }
        for(int i = 0; i < words.size(); i++)
        {
            flag = 1;
            for(int j = 0; j < words[i].length(); j++)
            {
                if(hashset.find(words[i][j]) == hashset.end())
                {
                    flag = 2;
                    break;
                }
            }
            if(flag == 1)
            {
                consistent++;
            }

        }

        return consistent;
        
    }
};

File: 1786-count-the-number-of-consistent-strings/1786-count-the-number-of-consistent-strings.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {

        int cnt = 0;
        Set<Character> st = new HashSet<Character>();

        for(int i = 0; i < allowed.length(); i++)
        {
            st.add(allowed.charAt(i));
        }

       for(String word : words)
       {
        boolean flag = true;
        for(int j = 0; j < word.length(); j++)
        {

        // if(!allowed.contains(String.valueOf(words[i].charAt(j))))
        if(!st.contains(word.charAt(j)))
        {
            flag = false;
            break;
        }

        }

        if(flag)
        {
            cnt++;
        }
       } 

       return cnt;
    }
}